scripts/topology_generator/fat_tree.py

```diff
@@ -0,0 +1,125 @@
+import yaml
+
+def add_bidirectional_link(config, from_node, to_node, link_counter, preset = "default"):
+    config["links"][f"link{link_counter}"] = {
+        "from": from_node, "to": to_node, "preset-name": preset
+    }
+    link_counter += 1
+    config["links"][f"link{link_counter}"] = {
+        "from": to_node, "to": from_node, "preset-name": preset
+    }
+    return link_counter + 1
+
+def generate_fat_tree_config(k):
+    if k % 2 != 0 or k < 2:
+        raise ValueError("k must be an even integer >= 2")
+    
+    num_pods = k
+    edge_per_pod = k // 2
+    aggr_per_pod = k // 2
+    hosts_per_pod = edge_per_pod * (k // 2)
+    core_switches = (k // 2) ** 2
+    total_hosts = (k ** 3) // 4
+    senders = total_hosts // 2
+    receivers = total_hosts // 2
+    
+    config = {
+        "presets": {
+            "link": {
+                "default": {
+                    "latency": "1ns",
+                    "throughput": "1Gbps",
+                    "ingress_buffer_size": "4096B",
+                    "egress_buffer_size": "4096B"
+                },
+                "medium_speed": {
+                    "latency": "1ns",
+                    "throughput": "10Gbps",
+                    "ingress_buffer_size": "4096B",
+                    "egress_buffer_size": "4096B"
+                },
+                "high_speed": {
+                    "latency": "1ns",
+                    "throughput": "100Gbps",
+                    "ingress_buffer_size": "4096B",
+                    "egress_buffer_size": "4096B"
+                }
+            },
+            "switch": {
+                "edge-preset": {
+                    "ecn": {"min": 0.2, "max": 0.3, "probability": 0.5}
+                },
+                "aggr-preset": {
+                    "ecn": {"min": 0.2, "max": 0.3, "probability": 0.7}
+                },
+                "core-preset": {
+                    "ecn": {"min": 0.2, "max": 0.3, "probability": 1.0}
+                }
+            }
+        },
+        "packet-spraying": {"type": "ecmp"},
+        "hosts": {},
+        "switches": {},
+        "links": {}
+    }
+    
```

Comment:
I think it would be nice to simplify hosts and switches naming. Also due to fact it used many times, creating lambdas for them makes sense
```suggestion
    host_name = labmda pod_idx, host_idx: f"pod{pod_idx}_host{host_idx}"
    arrg_name = lambda pod_idx, arrg_idx : f"pod{pod_idx}_arrg{aggr_idx}"
    edge_name = lambda pod_idx, edge_idx : f"pod{pod_idx}_arrg{edge_idx}"
```